be/src/storage/lake/delta_writer.cpp

```diff
@@ -353,6 +420,69 @@ Status DeltaWriterImpl::finish(DeltaWriter::FinishMode mode) {
     return Status::OK();
 }
 
+Status DeltaWriterImpl::_fill_auto_increment_id(const Chunk& chunk) {
+    ASSIGN_OR_RETURN(auto tablet, _tablet_manager->get_tablet(_tablet_id));
+
+    // 1. get pk column from chunk
+    vector<uint32_t> pk_columns;
+    for (size_t i = 0; i < _tablet_schema->num_key_columns(); i++) {
+        pk_columns.push_back((uint32_t)i);
+    }
+    Schema pkey_schema = ChunkHelper::convert_schema(*_tablet_schema, pk_columns);
+    std::unique_ptr<Column> pk_column;
+    if (!PrimaryKeyEncoder::create_column(pkey_schema, &pk_column).ok()) {
+        CHECK(false) << "create column for primary key encoder failed";
+    }
+    auto col = pk_column->clone();
+
+    PrimaryKeyEncoder::encode(pkey_schema, chunk, 0, chunk.num_rows(), col.get());
+    std::vector<std::unique_ptr<Column>> upserts;
+    upserts.resize(1);
+    upserts[0] = std::move(col);
+
+    std::vector<uint64_t> rss_rowid_map(upserts[0]->size(), (uint64_t)((uint32_t)-1) << 32);
+    std::vector<std::vector<uint64_t>*> rss_rowids;
+    rss_rowids.resize(1);
+    rss_rowids[0] = &rss_rowid_map;
+
+    // 2. probe index
+    int64_t version = tablet.get_version();
+    auto res = tablet.get_metadata(version);
```

Comment:
`tablet_mgr->get_latest_cached_tablet_metadata` can get tablet meta from cache, which is more faster.

Reply:
done